### PR TITLE
DMS - Add Agent Jobs and Login migration for SQL to MI public preview scenario

### PR DESCRIPTION
**Pull request by hitenjava**
REST Spec - https://github.com/Azure/azure-rest-api-specs/tree/master/specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-03-31-preview

PR - https://github.com/Azure/azure-rest-api-specs/commit/f8e623cef86af9d86649cc44ba36041c3133bd5d

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/DataMigration/DataMigration-ReRecordAfterCRUpdate.playlist [hunk 1]
```diff
@@ -1 +0,0 @@
-<Playlist Version="1.0"><Add Test="DataMigration.Tests.ScenarioTests.CRUDServiceTests.DeleteResourceSucceeds" /></Playlist>
```
Comment (dsgouda) [inline src/SDKs/DataMigration/DataMigration-ReRecordAfterCRUpdate.playlist]: Was this intentional?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Left a few comments

File: src/SDKs/DataMigration/DataMigration-ReRecordAfterCRUpdate.playlist (hunk 1 above)
Comment (hitenjava) [reply src/SDKs/DataMigration/DataMigration-ReRecordAfterCRUpdate.playlist]: @shahabhijeet sent out an email about tests with this file will have to be re-recorded.
I have re-recorded all the tests and so I assumed it should be safe to remove this file.

File: src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj [hunk 2]
```diff
@@ -6,7 +6,7 @@
 	<PropertyGroup>
 		<PackageId>Microsoft.Azure.Management.DataMigration</PackageId>
 		<Description>Provides developers with libraries for the Database Migration Service (DMS) under Azure Resource manager to deploy an instance of DMS. Create, Delete, Create project, Start Migration, and more.</Description>
```
[the remark this answers is not in the document]
Comment (hitenjava) [reply src/SDKs/DataMigration/Management.DataMigration/Microsoft.Azure.Management.DataMigration.csproj]: Done

File: src/SDKs/_metadata/datamigration_resource-manager.txt [hunk 3]
```diff
@@ -1,9 +1,9 @@
-2018-03-02 20:43:19 UTC
+2018-03-29 22:42:40 UTC
```
[the remark this answers is not in the document]
Comment (hitenjava) [reply src/SDKs/_metadata/datamigration_resource-manager.txt]: Done

File: src/SDKs/DataMigration/Management.DataMigration/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -7,8 +7,8 @@
 [assembly: AssemblyTitle("Microsoft Azure Data Migration Management Library")]
 [assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Data Migration Resources.")]
 
-[assembly: AssemblyVersion("0.2.0")]
-[assembly: AssemblyFileVersion("0.2.0")]
+[assembly: AssemblyVersion("0.3.0")]
+[assembly: AssemblyFileVersion("0.3.0")]
```
[the remark this answers is not in the document]
Comment (hitenjava) [reply src/SDKs/DataMigration/Management.DataMigration/Properties/AssemblyInfo.cs]: Done

File: src/SDKs/DataMigration/DataMigration-ReRecordAfterCRUpdate.playlist (hunk 1 above)
Comment (dsgouda) [reply src/SDKs/DataMigration/DataMigration-ReRecordAfterCRUpdate.playlist]: @shahabhijeet would it be safe to remove this file?

**Comment by hitenjava** [comment]
Addressed all the comments.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please run `msbuild build.proj /t:build /p:SDKs\DataMigration`
This should modify the .props file, please commit the change.

File: src/SDKs/DataMigration/AzSdk.RP.props [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <!--This file and it's contents are updated at build time moving or editing might result in build failure. Take due deligence while editing this file-->
   <PropertyGroup>
-    <AzureApiTag>DataMigration_2018-03-15-preview;</AzureApiTag>
+    <AzureApiTag>DataMigration_2018-03-31-preview;DataMigration_2018-03-15-preview;</AzureApiTag>
```
Comment (dsgouda) [inline src/SDKs/DataMigration/AzSdk.RP.props]: Apologize for the confusion, the earlier commit was correct, just needs one tag.

**Review by dsgouda — APPROVED** [review]
LGTM subject to builds passing.

**Comment by hitenjava** [comment]
@dsgouda Build passed. Please feel free to merge.
Thanks!

**Pull request merged**